src/diffusers/models/cross_attention.py

```diff
@@ -436,6 +436,49 @@ def __call__(self, attn: CrossAttention, hidden_states, encoder_hidden_states=No
         return hidden_states
 
 
+class TorchAttentionProcessor:
+    def __init__(self):
+        if not hasattr(F, "scaled_dot_product_attention"):
+            raise ImportError(
+                "TorchAttentionProcessor requires PyTorch 2.0, to use it, please upgrade PyTorch to 2.0."
+            )
+
+    def __call__(self, attn: CrossAttention, hidden_states, encoder_hidden_states=None, attention_mask=None):
+        batch_size, _, inner_dim = hidden_states.shape
+
+        if attention_mask is not None:
+            raise NotImplementedError("Attention mask is not supported yet.")
```

Comment:
Is it not supported because it cannot be supported or because it's difficult?

Reply:
It is indeed supported, but for now, when `attention_mask` is passed `scaled_dot_product_attention` calls the vanilla c++ attention and not the `memory_efficient_attention`.  Maybe @HamidShojanazeri could comment on when the `scaled_dot_product_attention` will support `attention_mask` for efficient kernels.

Reply:
But that's fine no? Maybe we can in this case just throw a warning but still run the code? 

IMO, we should aim for 1-to-1 feature parity with our default `CrossAttentionProcessor` and always use `TorchAttentionProcessor` by default when `torch.__version__ >= 2.0.0` no?